### DISCUSSION TITLE
About CUDAToolkit_ROOT variable during installation from source

### DISCUSSION THREAD

**Discussion by scott-5** [deepmodeling/deepmd-kit#4362] [Q&A] [answered]
I can successfully install deepmd-kit in an environment as the following:
```
pip install torch torchvision torchaudio
pip install git+https://github.com/deepmodeling/deepmd-kit@v3.0.0rc0
```
But I noticed [CUDAToolkit_ROOT](https://docs.deepmodeling.com/projects/deepmd/en/latest/install/install-from-source.html#envvar-CUDAToolkit_ROOT) hint that "NVCC is required". I cannot find it through `which nvcc`, it seems that the above command does not install NVCC (or CUDAToolkit).
So what I want to ask is that NVCC is not necessary? Or the above installation commands may be incomplete, do I still need to install CUDAToolkit?

My purpose of building from source is to compare the performance (such as training speed) between installation from conda (online or offline) and building from source code.
If anyone already has the answer, perhaps I can skip this comparison step. I would greatly appreciate your response.

**Comment by njzjz** [comment, marked as answer]
You need to set [`DP_VARIANT`](https://docs.deepmodeling.com/projects/deepmd/en/latest/install/install-from-source.html#envvar-DP_VARIANT) to `cuda` and [`DP_ENABLE_PYTORCH`](https://docs.deepmodeling.com/projects/deepmd/en/latest/install/install-from-source.html#envvar-DP_ENABLE_PYTORCH) to `1` if you need to run DPA-2 in LAMMPS or compress a model, and then the nvcc is required. Otherwise it's not necessary.

**Reply by scott-5** [reply]
Thank you for your reply. Indeed, the nvcc is required after setting those variable. 
I have another question. In order to get the best performance, which installation method is most recommended for deepmd-kit, lammps, and plumed?

**Reply by njzjz** [reply]
I never tested it, but I don't expect there to be a significant difference.
